azure-kusto-data/azure/kusto/data/aio/client.py

```diff
@@ -0,0 +1,87 @@
+    async def __aenter__(self) -> "KustoClient":
+        return self
+
+    def __aexit__(self, exc_type, exc_val, exc_tb):
+        self._session.__aexit__(exc_type, exc_val, exc_tb)
+
+    @aio_documented_by(KustoClientSync.execute)
+    async def execute(self, database: str, query: str, properties: ClientRequestProperties = None) -> KustoResponseDataSet:
+        query = query.strip()
+        if query.startswith("."):
+            return await self.execute_mgmt(database, query, properties)
+        return await self.execute_query(database, query, properties)
+
+    @aio_documented_by(KustoClientSync.execute_query)
+    async def execute_query(self, database: str, query: str, properties: ClientRequestProperties = None) -> KustoResponseDataSet:
+        return await self._execute(self._query_endpoint, database, query, None, KustoClient._query_default_timeout, properties)
+
+    @aio_documented_by(KustoClientSync.execute_mgmt)
+    async def execute_mgmt(self, database: str, query: str, properties: ClientRequestProperties = None) -> KustoResponseDataSet:
+        return await self._execute(self._mgmt_endpoint, database, query, None, KustoClient._mgmt_default_timeout, properties)
+
+    @aio_documented_by(KustoClientSync.execute_streaming_ingest)
+    async def execute_streaming_ingest(
+        self,
+        database: str,
+        table: str,
+        stream: io.IOBase,
+        stream_format: Union[DataFormat, str],
+        properties: ClientRequestProperties = None,
+        mapping_name: str = None,
+    ):
+        stream_format = stream_format.value if isinstance(stream_format, DataFormat) else DataFormat(stream_format.lower()).value
+        endpoint = self._streaming_ingest_endpoint + database + "/" + table + "?streamFormat=" + stream_format
+        if mapping_name is not None:
+            endpoint = endpoint + "&mappingName=" + mapping_name
+
+        await self._execute(endpoint, database, None, stream, self._streaming_ingest_default_timeout, properties)
+
+    @aio_documented_by(KustoClientSync._execute)
+    async def _execute(
+        self, endpoint: str, database: str, query: Optional[str], payload: Optional[io.IOBase], timeout: timedelta, properties: ClientRequestProperties = None
+    ) -> KustoResponseDataSet:
+
+        request_params = ExecuteRequestParams(database, payload, properties, query, timeout, self._request_headers)
+        json_payload = request_params.json_payload
+        request_headers = request_params.request_headers
+        timeout = request_params.timeout
+
+        if self._auth_provider:
+            request_headers["Authorization"] = await self._auth_provider.acquire_authorization_header_async()
+
+        async with self._session.post(endpoint, headers=request_headers, data=payload, json=json_payload, timeout=timeout.seconds) as response:
+            response_json = await response.json()
+
+            if response.status != 200:
+                raise KustoServiceError([response_json], response)
```

Comment:
In the sync client you create it as follows `KustoServiceError(response_json, response)`